### PR TITLE
Sec: Fix under constrained selector

### PR DESCRIPTION
**Pull request by 0xJepsen**
Addresses an under constrained `Selector` circuit (Pointed out by @lonerapier), fixes a broken import path, and addresses a broken nightly clippy.

### CODE REVIEW

File: circuits/aes-gcm/utils.circom [hunk 1]
```diff
@@ -403,15 +382,20 @@ template Contains(n) {
...
-    signal input index;
+    signal input index; 
     signal output out[n];
     assert(index >= 0 && index < m);
 
     signal selector[m];
+    component Equal[m];
     for (var i = 0; i < m; i++) {
-        selector[i] <-- index == i ? 1 : 0;
+        Equal[i] = IsEqual();
```
Comment (lonerapier) [inline circuits/aes-gcm/utils.circom:395]: if it helps, we can do this is a single line using [anonymous components](https://docs.circom.io/circom-language/anonymous-components-and-tuples/#the-use-of-for-signal-arrays).

```
selector[i] <== IsEqual()([index, i]);
```

Comment (0xJepsen) [reply circuits/aes-gcm/utils.circom]: This is cool! Made the switch, it's for sure a little cleaner

File: circuits/aes-gcm/utils.circom [hunk 2]
```diff
@@ -403,15 +382,17 @@ template Contains(n) {
...
+    signal input index; 
     signal output out[n];
     assert(index >= 0 && index < m);
 
     signal selector[m];
+    component Equal[m];
     for (var i = 0; i < m; i++) {
-        selector[i] <-- index == i ? 1 : 0;
+        selector[i] <== IsEqual()([index, i]);
         selector[i] * (1 - selector[i]) === 0;
```
Comment (lonerapier) [inline circuits/aes-gcm/utils.circom:396]: i don't think we need this constraint now, IsEqual already has that

Comment (0xJepsen) [reply circuits/aes-gcm/utils.circom:396]: The one on line 415?

Comment (lonerapier) [reply circuits/aes-gcm/utils.circom:396]: yeah

**Review by lonerapier — APPROVED** [review]
LFG!! 🚀

**Pull request merged**